File: sdk/core/src/test/java/com/contextsolutions/securegateway/core/JvmSodiumProvider.java
package com.contextsolutions.securegateway.core;

import com.goterl.lazysodium.LazySodium;
import com.goterl.lazysodium.LazySodiumJava;
import com.goterl.lazysodium.SodiumJava;

/**
 * Desktop/JVM {@link SodiumProvider} for {@code :core}'s own unit tests (which exercise
 * {@link Crypto} directly). Registered via {@code META-INF/services} in test resources
 * only — never in {@code :core} main, so the Android AAR's classpath never sees it.
 */
public final class JvmSodiumProvider implements SodiumProvider {

    @Override
    public LazySodium get() {
        return new LazySodiumJava(new SodiumJava());
    }
}

File: sdk/android/src/test/kotlin/com/contextsolutions/securegateway/mobile/JvmSodiumProvider.kt
package com.contextsolutions.securegateway.mobile

import com.goterl.lazysodium.LazySodium
import com.goterl.lazysodium.LazySodiumJava
import com.goterl.lazysodium.SodiumJava
import com.contextsolutions.securegateway.core.SodiumProvider

/**
 * Desktop/JVM [SodiumProvider] for the mobile SDK's JVM unit tests
 * (`VectorsConformanceKtTest` runs the shared [com.contextsolutions.securegateway.core.Crypto]). Registered
 * via `META-INF/services` in test resources only — the real Android binding ships in the
 * `:android-aar` module, never here.
 */
class JvmSodiumProvider : SodiumProvider {
    override fun get(): LazySodium = LazySodiumJava(SodiumJava())
}
